### PR TITLE
Audio devices split

### PR DESCRIPTION
**Pull request by vladest**
https://highfidelity.fogbugz.com/f/cases/edit/6885/Split-Desktop-and-VR-Audio-Settings

Test plan:
- open Tablet in Desktop mode
- click Audio button
- try to switch audio devices for both Desktop/VR contexts
- make sure it works
- switch to VR/HMD
- make sure it switched to device, selected previously in Desktop mode
- repeat pp. 2-4
- Make sure audio settings are saved between sessions

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2468/HighFidelity-Beta-PR11182-b22745d57a90df73b54f8d967e04c0f3fd6d92f4.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2468/HighFidelity-Beta-PR11182-b22745d57a90df73b54f8d967e04c0f3fd6d92f4.exe

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

open Tablet in Desktop mode **Check**
click Audio button **Check**
try to switch audio devices for both Desktop/VR contexts **Check**
make sure it works **Check**
switch to VR/HMD **Check**
make sure it switched to device, selected previously in Desktop mode **Check**
repeat pp. 2-4 **Check**
Make sure audio settings are saved between sessions **Check**

Looks good to me! :) **Pass**

**Comment by anshuman64** [comment]
I'm testing this one more thoroughly...

**Comment by anshuman64** [comment]
Audio settings weren't saved between sessions.

Repro: 
1. After clean install, these were my audio settings:
audio1
2. I closed Interface and reopened it. My audio settings changed to:
audio2

Expected behavior is that they are maintained in between sessions.

**Comment by YuppyPlays** [comment]
@anshuman64 
Strange, possibly because of difference in headset but mine retained the state between interface sessions.

Good find!

**Comment by anshuman64** [comment]
@vladest Let's pause on this PR for now. The designers are rethinking the design. We might have to close this PR and try again.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2572/HighFidelity-Beta-PR11182-1a32a6fac39a2916ec729dd7f9630edc97d052d3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2572/HighFidelity-Beta-PR11182-1a32a6fac39a2916ec729dd7f9630edc97d052d3.exe

**Comment by anshuman64** [comment]
@vladest I tested this. The audio dialog is coming up blank. It is blank grey in desktop mode. In tablet mode, it is blank black. Please check.

Also, I saw your comment on the FB ticket: we still want the audio bars.

**Comment by vladest** [comment]
@anshuman64 can you please give me logs after Audio gets opened? thanks

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2578/HighFidelity-Beta-PR11182-f77b913a143edeac05b8987cb8b2aa225cb9f0ca.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2578/HighFidelity-Beta-PR11182-f77b913a143edeac05b8987cb8b2aa225cb9f0ca.dmg

**Comment by anshuman64** [comment]
[hifi-log.txt](https://github.com/highfidelity/hifi/files/1232408/hifi-log.txt)

Relevant lines:

[08/17 12:37:32] [WARNING] [default] file:///C:/Program Files/High Fidelity - PR11182/resources//qml/hifi/audio/Audio.qml:59:9: Type AudioControls.AudioTabButton unavailable
[08/17 12:37:32] [WARNING] [default] file:///C:/Program Files/High Fidelity - PR11182/resources//qml/hifi/audio/AudioTabButton.qml: No such file or directory
[08/17 12:37:32] [DEBUG] [hifi.controllers] Attempting to  disable  mapping  "file:///C:/Program Files/High Fidelity - PR11182/scripts/system/pal.js-click"
[08/17 12:37:32] [DEBUG] [hifi.controllers] Attempting to  disable  mapping  "file:///C:/Program Files/High Fidelity - PR11182/scripts/system/pal.js-press"

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2579/HighFidelity-Beta-PR11182-855f9d8a950481d6b583129b8ff50c09a2b35766.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2579/HighFidelity-Beta-PR11182-855f9d8a950481d6b583129b8ff50c09a2b35766.exe

**Comment by anshuman64** [comment]
@vladest Put some comments in the ticket: https://highfidelity.fogbugz.com/f/cases/6885/Split-Desktop-and-VR-Audio-Settings

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2666/HighFidelity-Beta-PR11182-f7a704d9f43468cdf4958f9a010d47e79ebdc3b4.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2666/HighFidelity-Beta-PR11182-f7a704d9f43468cdf4958f9a010d47e79ebdc3b4.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2832/HighFidelity-Beta-PR11182-78aa5820d06b9fbe6e4a20bdd9fc5ff08bea9e6d.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2832/HighFidelity-Beta-PR11182-78aa5820d06b9fbe6e4a20bdd9fc5ff08bea9e6d.dmg

**Comment by anshuman64** [comment]
@vladest the audio switching saves now! Great!

Asking designer to review and then will remove Do Not Merge tag

**Comment by anshuman64** [comment]
audioappv4

Minor changes:
- {Mute Microphone, Enable noise reduction, Show audio level meter} should be moved to the right so that Mute Microphone and Enable Noise Reduction are in line with "Choose Input Device"
- Audio meter for input devices should be in the middle of the line of text for input device

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3082/HighFidelity-Beta-PR11182-e17f05a33764af13fae631b13f3bc2e56ccc3e12.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3082/HighFidelity-Beta-PR11182-e17f05a33764af13fae631b13f3bc2e56ccc3e12.exe

**Comment by anshuman64** [comment]
Looks good to me! Confirming with designer...

**Comment by cain-kilgore** [comment]
Testing.

**Comment by Nex-Pro** [comment]
testing this

**Comment by anshuman64** [comment]
Both of you can do it. Rigorously test switching devices

**Comment by Nex-Pro** [comment]
## FAIL device listed twice and can be selected twice.

See image.

hifi-snap-by--on-2017-09-07_18-02-46

**Comment by cain-kilgore** [comment]
# Test Passed
Test | Desktop | HMD
--- | --- | ---
open Tablet in Desktop mode | Pass | 
click Audio button | Pass | 
try to switch audio devices for both Desktop/VR contexts | Pass
make sure it works | Pass
switch to VR/HMD | | Pass
make sure it switched to device, selected previously in Desktop mode | | Pass
repeat pp. 2-4 | | Pass
Make sure audio settings are saved between sessions | Pass | Pass

All good from here - I wasn't able to replicate the duplicate output device @Nex-Pro was having.

###### Intel i5 7600K @ 4Ghz, 16GB DDR4 2133Mhz Corsair Vengeance, Asus Strix GTX 1050Ti 3GB, HTC Vive

**Comment by anshuman64** [comment]
@Nex-Pro That didn't occur in master branch?

**Comment by Nex-Pro** [comment]
@anshuman64 Those two identically named outputs get selected with one click in latest dev build.

**Comment by anshuman64** [comment]
build this please

**Comment by anshuman64** [comment]
build this please

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3427/HighFidelity-Beta-PR11182-6297448e907660bf508db14ff2f635775346c9e7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3427/HighFidelity-Beta-PR11182-6297448e907660bf508db14ff2f635775346c9e7.exe

**Comment by vladest** [comment]
please do not test for a while until further notice. there will be more commits

**Comment by vladest** [comment]
now its ready for testing

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3442/HighFidelity-Beta-PR11182-10f7f081713e491549bb9d5ceb4471cc262f98ef.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3442/HighFidelity-Beta-PR11182-10f7f081713e491549bb9d5ceb4471cc262f98ef.exe

**Comment by anshuman64** [comment]
@vladest Almost everything is looking good. Only bug I found: audio devices can be "deselected" by pressing the button again. It should work as in master where devices can't be deselected, but only switched between.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3452/HighFidelity-Beta-PR11182-7a36894f76e9f17b754e4e2c4e6aab9961769739.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3452/HighFidelity-Beta-PR11182-7a36894f76e9f17b754e4e2c4e6aab9961769739.dmg

File: interface/resources/qml/controls-uit/CheckBox2.qml [hunk 1]
```diff
@@ -0,0 +1,113 @@
+//
+//  CheckBox2.qml
+//
```
Comment (druiz17) [inline interface/resources/qml/controls-uit/CheckBox2.qml]: what is the difference between checkBox vs checkBox2. I think it is best to give checkBox2 a different name

Comment (vladest) [reply interface/resources/qml/controls-uit/CheckBox2.qml]: its based on qt quick controls 2
what name you suggest, taking in account we have to keep old name for compatibility reasons

Comment (druiz17) [reply interface/resources/qml/controls-uit/CheckBox2.qml]: Just a name so people know the difference of the two files. I am not quite sure what name to use. What new thing does qt quick controls 2 bring to checkboxes?

Comment (vladest) [reply interface/resources/qml/controls-uit/CheckBox2.qml]: more lightweight. faster creation, less memory footprint, faster

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3481/HighFidelity-Beta-PR11182-c48aba423ac7c168377d3c73fac0e5bca8155d4b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3481/HighFidelity-Beta-PR11182-c48aba423ac7c168377d3c73fac0e5bca8155d4b.exe

**Pull request merged**